File: RuletaRusa.java
import java.io.*;
import java.lang.Math;
import java.util.Random;

public class RuletaRusa{
	
	private String nombre;
	private int edad;
	private String estado = "Vivo";
	int i;
	
	InputStreamReader stream = new InputStreamReader(System.in);
	BufferedReader reader = new BufferedReader(stream);
	
	public void pedirDatos() throws IOException{
		System.out.println("********** <Bienvenido a la Ruleta Rusa> **********");
		System.out.println("--Ingrese sus datos");
		System.out.print("Ingrese el nombre: ");
		nombre = reader.readLine();
		System.out.print("Ingrese la edad: ");
		edad = Integer.valueOf(reader.readLine()).intValue();
	}
	
	private void comprobarEdad(){
		if(edad >= 18)
			System.out.println("\nPuede ingresar al juego, es mayor de edad.");
		else
			System.out.println("\nNo puede ingresar al juego, es menor de edad.");
	}
	
	public void disparar(){
		int numero = 0;
		comprobarEdad();
		if(edad >= 18){
			for(i=0; i<=6; i++)
				numero = (int)(Math.random()*6+1);
			
			System.out.println("\nLa bala salio del numero: " + numero);
			
			if(numero == 4){
				estado = "Muerto";
				System.out.println("El jugador esta: " + estado);
			}
		}
	}
	
	public void mostrarDatos(){
		System.out.println("\n********** -Estado actual del jugador- **********");
		System.out.println("Nombre: " + nombre);
		System.out.println("Edad: " + edad);
		System.out.println("Estado: " + estado);
	}
	
	public void setNombre(String nombre){
		this.nombre = nombre;
	}
	
	public void setEdad(int edad){
		this.edad = edad;
	}
	
	public void setEstado(String estado){
		this.estado = estado;
	}
	
	public String getNombre(){
		return nombre;
	}
	
	public int getEdad(){
		return edad;
	}
	
	public String getEstado(){
		return estado;
	}
	
}

File: C++/Ruleta rusa.cpp
#include<iostream>
#include<stdlib.h>
#include<time.h>
using namespace std;

class Jugador{
	private:
		string nombre;
		int edad;
		string estado;

	public:
		//métodos set()
		void setEstado(string);
		//métodos get()
		string getNombre();
		int getEdad();
		string getEstado();
		//métodos de la clase
		void pedirDatos(string, int);
		void comprobarEdad();
		void disparar();
		void mostrarDatos();
};

void Jugador::pedirDatos(string _nombre, int _edad){
	nombre = _nombre;
	edad = _edad;
	estado = "Vivo";
}

void Jugador::comprobarEdad(){
	if(Jugador::getEdad() >= 18){
		cout << "Podra entrar al juego" << endl;
	}
	else{
		cout << "Es menor de edad, no puede jugar" << endl;
	}
}

void Jugador::disparar(){
	int num, i;
	srand(time(NULL));
	
	if(Jugador::getEdad() >= 18){
		for(i=1; i<=6; i++){
			num = 1 + rand() % (7-1);
		}
		cout << "Numero: " << num << endl;
		if(num == 4){
			estado = "muerto";
		}
	}
}

void Jugador::mostrarDatos(){
	cout << "\n*****Datos del Jugador*****" << endl;
	cout << "Nombre: " << nombre << endl;
	cout << "Edad:  " << edad << endl;
	cout << "Estado: " << estado << endl;
}

string Jugador::getNombre(){
	return nombre;
}

int Jugador::getEdad(){
	return edad;
}

string Jugador::getEstado(){
	return estado;
}

int main(){
	string nombreJugador;
	int edadJugador;
	Jugador obj1;
	
	cout << "********** Bienvenido a la ruleta rusa **********" << endl;
	cout << "Da enter para ingresar los datos del jugador: " << endl;
	system("pause");
	cout << "Ingrese el nombre: ";
	cin >> nombreJugador;
	cout << "Ingrese la edad del jugador: ";
	cin >> edadJugador;
	obj1.pedirDatos(nombreJugador, edadJugador);
	obj1.comprobarEdad();
	obj1.disparar();
	obj1.mostrarDatos();
	
	system("pause");
	return 0;
}
